### PR TITLE
BREAKING implement (*Backend).Delete method per #1

### PR DESCRIPTION
**Pull request by ahouene**
(*Backend).Delete has been implemented for all three existing backends. Corresponding tests have been added.

### CODE REVIEW

File: backends/fs/fs.go [hunk 1]
```diff
@@ -78,6 +78,10 @@ func (b *Backend) Store(ctx context.Context, name string, data []byte) error {
 	return os.Rename(tmpPath, fullPath)
 }
 
+func (b *Backend) Delete(ctx context.Context, name string) error {
+    return os.Remove(filepath.Join(b.rootPath, name))
```
Comment (wojas) [inline backends/fs/fs.go:82]: Security: missing allowedName call.

**Comment by wojas** [comment]
One remaining design issue is what to do if we could not delete the object because it does not exist.

The current implementation returns `ErrNotExist` in such case, but I think I would prefer to not return any error in such a case. All the other methods are idempotent, allowing you to repeat the same call without getting different results. Returning `ErrNotExist` breaks this pattern. [HTTP DELETE is also idempotent](https://datatracker.ietf.org/doc/html/rfc7231#section-4.2.2).

Not returning an error in this case also makes the API easier to use. I think most code that consumes this API would end up checking for this specific error and ignoring it.  I cannot think of a case where getting an `ErrNotExist` is useful. If you really care if it existed, you could do a List before the call.

**Comment by ahouene** [comment]
Good point and insight, I'm fixing this!

File: plugins.go [hunk 2]
```diff
@@ -14,10 +14,14 @@ import (
...
 // optional interfaces that a backend can implement.
 type Interface interface {
+	// List retrieves a BlobList with given prefix.
 	List(ctx context.Context, prefix string) (BlobList, error)
+	// Load brings a whole value, chosen by name, into memory.
 	Load(ctx context.Context, name string) ([]byte, error)
+	// Store sends value to storage for a given name. 
 	Store(ctx context.Context, name string, data []byte) error
-	// FIXME: add Delete and document each method
+	// Delete entry, identified by name, from storage
```
Comment (wojas) [inline plugins.go:23]:
```suggestion
	// Delete entry, identified by name, from storage. No error is returned if it does not exist.
```

**Review by wojas — APPROVED** [review]
LGTM! We will need to release a v0.2.0 with this change.

**Pull request merged**